ray-on-gke/tpu/kuberay-tpu-webhook/webhook_main_test.go

```diff
@@ -1315,9 +1189,63 @@ func Test_GetEnvironmentVariable(t *testing.T) {
 	}
 }
 
+func Test_GetSliceToWorkerIDs(t *testing.T) {
+	setupTest(t)
+
+	tests := map[string]struct {
+		numOfHosts               int32
+		numReplicas              int
+		podsInGroup              []*corev1.Pod
+		expectedSliceToWorkerIDs map[slice][]int
+	}{
+		"getSliceToWorkerIDs with nil Pod list": {
+			// this can occur when no Pods with the Ray group name have been cached
+			// should return an empty mapping
+			podsInGroup:              nil,
+			expectedSliceToWorkerIDs: make(map[slice][]int),
+		},
+		"getSliceToWorkerIDs for with CPU pod list": {
+			// sliceToWorkerIDs should return an empty mapping
+			numOfHosts:               int32(1),
+			numReplicas:              1,
+			podsInGroup:              testCPUPods,
+			expectedSliceToWorkerIDs: make(map[slice][]int),
+		},
+		"getSliceToWorkerIDs for with TPU pod list": {
+			// sliceToWorkerIDs should be populated with TPU worker IDs
+			numOfHosts:  int32(2),
+			numReplicas: 2,
+			podsInGroup: testInterceptedTPUPods,
+			expectedSliceToWorkerIDs: map[slice][]int{
+				slice{instanceName, groupNameStr, namespaceStr, 0, int32(2)}: []int{0, 1},
+				slice{instanceName, groupNameStr, namespaceStr, 1, int32(2)}: []int{0, 1},
+			},
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			sliceToWorkerIDs, err := getSliceToWorkerIDs(tc.podsInGroup, instanceName, groupNameStr, namespaceStr, tc.numOfHosts)
+
+			// sliceToWorkerIDs should be populated with slices and unique TPU_WORKER_IDs for each Pod
+			assert.Equal(t, err, nil)
+			for slice, workerIDs := range tc.expectedSliceToWorkerIDs {
+				assert.Contains(t, sliceToWorkerIDs, slice)
+				assert.Equal(t, len(workerIDs), len(sliceToWorkerIDs[slice]))
+				sort.Ints(workerIDs)
+				for index, value := range workerIDs {
+					assert.Equal(t, sliceToWorkerIDs[slice][index], value)
+				}
+			}
+		})
+	}
+}
+
 func Test_MutatePod(t *testing.T) {
 	setupTest(t)
 
+	testPodLister := setupInformer(testTPUPods)
```

Comment:
Wondering if we should be initializing the pod lister per test case inside `t.Run`. Do we anticipate the state to change based on the test case?

Reply:
Done in [b70535d](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/b70535d96c495b0afafbfe07cd8bfa82a3d0d5b8), we now can change the # of Pods / type of Pods used in the PodLister.